background/handle-settings-page.js

```diff
@@ -24,24 +24,13 @@ chrome.runtime.onMessage.addListener(function (request, sender, sendResponse) {
     chrome.storage.sync.set({
       addonsEnabled: scratchAddons.localState.addonsEnabled,
     });
-
+    const manifest = scratchAddons.manifests.find((a) => a.addonId === addonId).manifest;
+    const { dynamicEnable, dynamicDisable } = manifest;
     // Fire disabled event for userscripts
-    // TODO: this might not be an addon being reenabled. We should consider this
-    // in case we want to provide userscripts with a way to run after the page
-    // has loaded, dynamically.
-    chrome.tabs.query({}, (tabs) =>
-      tabs.forEach(
-        (tab) =>
-          (tab.url || (!tab.url && typeof browser !== "undefined")) &&
-          chrome.tabs.sendMessage(tab.id, {
-            fireEvent: {
-              target: "self",
-              name: newState ? "reenabled" : "disabled",
-              addonId,
-            },
-          })
-      )
-    );
+    if (dynamicEnable && newState)
+      scratchAddons.localEvents.dispatchEvent(new CustomEvent("addonEnabled", { detail: { addonId, manifest } }));
+    if (dynamicDisable && !newState)
```

Comment:
Would prefer explicitly comparing newState, `newState === true` and `newState === false`, for clarity